identity/app/idapiclient/IdApi.scala

```diff
@@ -60,6 +62,18 @@ abstract class IdApi(val apiRootUrl: String, http: Http, jsonBodyParser: JsonBod
     val response = http.POST(apiUrl(apiPath), Some(updatedSavedArticles), params, headers)
     response map extractSavedArticles }
 
+  def hasNonDefaultSavedArticles(auth: Auth) = {
+    def hasNoSavedArticles(savedArticles: List[SavedArticle]) = {savedArticles.isEmpty}
+
+    def hasOnlyDefaultSavedArticle(savedArticles: List[SavedArticle]) =
+      savedArticles.length == 1 && savedArticles.head.shortUrl == "/p/4ab7x"
+
+    savedArticles(auth).map(_.fold(
+      {_ => true},
+      { prefs => hasNoSavedArticles(prefs.articles) || hasOnlyDefaultSavedArticle(prefs.articles)})
```

Comment:
:+1:

Also, best to always create (and name) boolean values the way round you intend to test them. In this case I'd recommend something like:

    hasSavedArticles && doesNotHaveOnlyDefaultSavedArticle

I'm sure you could dome up with a better name.

That said, this feels again like indirection that isn't adding much value. Can you not inline these functions? Especially true of `hasNoSavedArticles`. Just the name is as long as the implementation (`savedArticles.isEmpty`) to say nothing of the definition, signature, invocation etc.